flang/test/Semantics/OpenMP/interop-construct.f90

```diff
@@ -0,0 +1,25 @@
+!RUN: not %flang_fc1 -fopenmp -fopenmp-version=52 %s 2>&1 | FileCheck %s
+
+subroutine test_interop_01()
+  use omp_lib
+  integer(omp_interop_kind) :: obj
+  !CHECK: error: Each interop-var may be speciﬁed for at most one action-clause of each interop construct.
```

Comment:
```suggestion
  !ERROR: error: Each interop-var may be speciﬁed for at most one action-clause of each interop construct.
```

Reply:
I did check making the above mentioned changes with the test_errors.py file check but the issue seems to but for some reason the test_errors.py is not picking up the actual errors from the console log output.

Reply:
If I ran the above code with `! RUN: %python %S/../test_errors.py %s %flang -fopenmp -fopenmp-version=52
` , I am getting the bellow error:
`FAIL: Flang :: Semantics/OpenMP/interop-construct.f90 (1311 of 3042)
******************** TEST 'Flang :: Semantics/OpenMP/interop-construct.f90' FAILED ********************
Exit Code: 1

Command Output (stdout):
--
--- 
+++ 
@@ -0,0 +1,3 @@

expect at 7: error: Each interop-var may be speciﬁed for at most one action-clause of each interop construct.
expect at 15: error: Each interop-type may be speciﬁed at most once.
expect at 23: error: A depend clause can only appear on the directive if the interop-type includes targetsync

FAIL

--
Command Output (stderr):
--
RUN: at line 1: "/usr/bin/python3.10" /home/mcw/Documents/Swatheesh/llvm-project/flang/test/Semantics/OpenMP/../test_errors.py /home/mcw/Documents/Swatheesh/llvm-project/flang/test/Semantics/OpenMP/interop-construct.f90 /home/mcw/Documents/Swatheesh/build/bin/flang -fopenmp -fopenmp-version=52
+ /usr/bin/python3.10 /home/mcw/Documents/Swatheesh/llvm-project/flang/test/Semantics/OpenMP/../test_errors.py /home/mcw/Documents/Swatheesh/llvm-project/flang/test/Semantics/OpenMP/interop-construct.f90 /home/mcw/Documents/Swatheesh/build/bin/flang -fopenmp -fopenmp-version=52

--

********************
`

Reply:
Tip: GitHub uses Markdown format, you can use three ` to enclose code or output.
See, https://www.markdownguide.org/extended-syntax/#fenced-code-blocks for more features

Reply:
Share you share the output of `flang-new flang/test/Semantics/OpenMP/interop-construct.f90 -fopenmp -fopenmp-version=50`

Reply:
Paste the output in this format
```console
$ flang-new flang/test/Semantics/OpenMP/interop-construct.f90 -fopenmp -fopenmp-version=50
[...]
```

Reply:
Thanks for the tip @Thirumalai-Shaktivel , Please see the output mentioned below:

```flang-new llvm-project/flang/test/Semantics/OpenMP/interop-construct.f90 -fopenmp -fopenmp-version=50
flang-20: warning: OpenMP support in flang is still experimental [-Wexperimental-option]
error: Semantic errors in llvm-project/flang/test/Semantics/OpenMP/interop-construct.f90
error: Each interop-var may be speciﬁed for at most one action-clause of each interop construct.
error: Each interop-type may be speciﬁed at most once.
error: A depend clause can only appear on the directive if the interop-type includes targetsync
```

Reply:
Nothing else, is it? 
Doesn't it print the location information of the error?
Something like the following:
```console
$ flang-new x.f90
error: Semantic errors in x.f90
./x.f90:2:24: error: Must be a constant value
      integer :: a(10) = x
                         ^
```

Reply:
If no, then we have to source the interop construct in the parser.

Reply:
Yeah, it doesn't print the location information of the error. 
Can you please let me know where should I add the source information so that I can add it and test whether it is working fine? Thanks!

Reply:
If you are confused, ping me here. I can explain more.

Reply:
See, https://github.com/swatheesh-mcw/llvm-project/pull/1/files#r1886185039

Reply:
I am a little confused here, can you just explain a little bit about the source information?